monash-human-power/dashboard
# Add Boost Calibration card to V3 Boost webpage

## Description

This PR adds the UI for the boost calibration card on the `boost` webpage for the `v3 dashboard`. The boost calibration card would allow users to view the distance travelled and the calibrated distance travelled by the bike and also set the calibration value. This PR uses hardcoded values for the distance and calibrated distance values (this will be updated to use correct values in a future PR). However, the input form to set the calibration value is vertically integrated (i.e it sends the calibrated value inserted on the correct `MQTT` topic).

Note: you'll notice that a lot of the code looks similar and that's because I basically copied over code from other files that achieved similar stuff that I wanted 😂. 
## Screenshots

<img width="1278" alt="Screen Shot 2021-02-07 at 6 31 01 pm" src="https://user-images.githubusercontent.com/63642262/107163221-72956300-69fc-11eb-956b-ac5c87a835d2.png">


## Steps to Test

1) Run `yarn install` inside `client/`
2) Use `yarn start` to launch the dashboard webpage 
3) On a seperate terminal navigate to `server/` and run `yarn start` to start the server
4) Use 2 separate terminals to subscribe to `boost/calibrate` and `boost/calibrate/reset` using localhost, here's. quick copy paste stuff you can use: 
`mosquitto_sub -h localhost -t "boost/calibrate"`
`mosquitto_sub -h localhost -t "boost/calibrate/reset"`
5) Navigate to the `boost` page on the `v3 dashboard` and check that the UI looks good. Use the [Figma design](https://www.figma.com/file/HoZBPfj3QqhmbiDUyxn8Ag/Initial-Design?node-id=0%3A1) to compare. 
6) Use the calibration value input form and the `set` and `reset` buttons to verify that they work correctly by checking the subscribed mqtt topics.
7) You can also try to break the input form by entering random stuff and see whether that breaks the code or not.
